iis-common/src/main/java/eu/dnetlib/iis/common/report/ReportPigCountersResolver.java

```diff
@@ -0,0 +1,48 @@
+package eu.dnetlib.iis.common.report;
+
+import java.util.List;
+
+import org.apache.commons.lang3.StringUtils;
+
+import com.google.common.collect.Lists;
+
+import eu.dnetlib.iis.common.counter.PigCounters;
+import eu.dnetlib.iis.common.schemas.ReportParam;
+
+/**
+ * Resolver of {@link ReportParam}s from {@link PigCounters} that uses {@link ReportPigCounterMapping}.
+ * 
+ * @author madryk
+ */
+public class ReportPigCountersResolver {
+
+    //------------------------ LOGIC --------------------------
+    
+    /**
+     * Resolve {@link ReportParam}s from {@link PigCounters} using {@link ReportPigCounterMapping}s.
+     * Only counters that are present in {@link ReportPigCounterMapping}s will be resolved.
+     */
+    public List<ReportParam> resolveReportCounters(List<ReportPigCounterMapping> reportPigCountersMapping, PigCounters pigCounters) {
```

Comment:
Could we change the parameter order? I think it would be more natural to have pigCounters first. The method then could be read like this: "resolve report counters from pigCounters by using the given mapping".